sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala

```diff
@@ -315,15 +298,15 @@ class UnsupportedOperationsSuite extends SparkFunSuite with SQLHelper {
     // future.
     expectedMsgs = Seq("Complete"))
 
-  for (outputMode <- Seq(Append, Update, Complete)) {
+  for (outputMode <- Seq(Append, Complete)) {
```

Comment:
The reason to fail here is not because of following aggregation operator. It's just because mapGroupsWithState is not supported with append/complete mode.

* Append/Complete mode: `mapGroupsWithState is not supported with Complete output mode on a streaming DataFrame/Dataset`
* Update mode: `Detected pattern of possible 'correctness' issue due to global watermark. The query contains stateful operation ...`

That said, test cases here are effectively the same with above test cases, without aggregation.

If you have test case for update mode on mapGroupsWithState followed by aggregation, then it's OK to just remove this test case. Otherwise, let's just change this to test for Update mode.

Reply:
Thanks for the close look! Yes I just realized [there is indeed a test on mapGroupsWithState in Update mode](https://github.com/apache/spark/blob/255691553e2f8f531c81a1e5b5a81d8559d3ef7a/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala#L516), I'll just remove these tests.